src/api/install.ts

```diff
@@ -204,6 +204,12 @@ export async function install (maybeOpts: InstallOptions) {
 
     await installInContext(installType, specs, [], ctx, preferredVersions, opts)
 
+    if (opts.shamefullyFlatten && specs.length > 0) {
+      await installPkgs(specs.map(spec => spec.raw), Object.assign({},
+        opts, {lock: false, reinstallForFlatten: true, update: false}
```

Comment:
you can use the spread operator

```js
{...opts, lock: false, reinstallForFlatten: true, update: false}
```

also, why is it needed? After uninstall it makes sense but in this case, it is a nameless installation and you make a named install after each nameless installation?